tests/unittest_nodes.py

```diff
@@ -626,6 +627,12 @@ def _test(self, value: Any) -> None:
         self.assertIs(node.value, value)
         self.assertTrue(node._proxied.parent)
         self.assertEqual(node._proxied.root().name, value.__class__.__module__)
+        with self.assertRaises(AttributeError):
+            with pytest.warns(DeprecationWarning) as warnings:
+                node.statement()
+                assert len(warnings) == 1
```

Comment:
```suggestion
        with (
            self.assertRaises(AttributeError),
            pytest.warns(DeprecationWarning) as records
        ):
            node.statement()
            assert len(records) == 1
```

Reply:
`black` failed on these combined `with` statements. It seems like it can't parse those statements together. I reverted in the latest commit but kept using `records`.

Reply:
I realized that it also requires at least Python 3.9. So even if black would work, we couldn't use it.
For reference: https://github.com/psf/black/issues/1948

This would work, but looks rather ugly with black formatting.
```py
        with self.assertRaises(AttributeError), pytest.warns(
            DeprecationWarning
        ) as records:
            node.statement()
            assert len(records) == 1
```
Let's leave it like it is then.